### ISSUE TITLE
Field renaming does not work for serialization

### ISSUE THREAD

**Issue by oharaandrew314** [cbeust/klaxon#79]
For Klaxon `2.0.6`, consider the following test cases:

```
import com.beust.klaxon.Json
import com.beust.klaxon.Klaxon
import com.beust.klaxon.KlaxonException
import org.hamcrest.CoreMatchers
import org.junit.Assert
import org.junit.Test

class KlaxonTest {
    data class TestClass(@Json(name="first_name") val firstName: String)
    private val mapper = Klaxon()
    private val obj = TestClass("John")

    /**
     * Attempt to serialize an object with a renamed field.
     * The serialized object should contain the renamed field.
     */
    @Test
    fun serialize() {
        Assert.assertThat(
            mapper.toJsonString(obj),
            CoreMatchers.containsString("first_name")
        )
    }

    /**
     * Attempt to deserialize an object with a "first_name" field.
     */
    @Test
    fun deserializeFormatted() {
        val jv = "{\"first_name\":\"John\"}"
        Assert.assertThat(
            mapper.parse<TestClass>(jv),
            CoreMatchers.equalTo(obj)
        )
    }

    /**
     * Attempt to deserialize an object with a "firstName" field
     * This should fail.
     */
    @Test(expected=KlaxonException::class)
    fun deserializeNonFormatted() {
        val jv = "{\"firstName\":\"John\"}"
        mapper.parse<TestClass>(jv)
    }
}
```

The `serialize` test `FAILS`, and the other two pass.  It appears that field renaming is not honoured for serialization.

**Comment by cbeust** [comment]
Should be fixed in 2.0.7.

**Comment by oharaandrew314** [comment]
I see the fix now.  Thanks!
